matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -166,6 +166,8 @@ public MatchResult doWork(List<GPXEntry> gpxList) {
         // Compute all candidates first.
         // TODO: Generate candidates on-the-fly within computeViterbiSequence() if this does not
         // degrade performance.
+        // @kodonnell: it's not currently possible as that'd mean calling queryGraph.lookup
```

Comment:
Not sure if you meant me - but I think it requires calling queryGraph.lookup multiple times because we're wanting to get the candidates at each step. E.g. first step: get candidates, lookup, do viterbi next step; second step: get candidates, lookup, do viterbi next step, ... 

If we create multiple queryGraphs, then we'll get duplicate virtual edge IDs, etc., which might be a problem.